src/reporter.js

```diff
@@ -133,6 +135,57 @@ class Reporter {
         return this.getJobURL(sessionId);
     }
 
+    createSauceTestReport(session) {
+        const testRun = new TestRun();
+        testRun.status = session.passed ? Status.Passed : Status.Failed;
+
+        let suite, fixture;
+
+        for (const test of session.tests) {
+            suite = testRun.withSuite(test.specPath);
+            fixture = suite.withSuite(test.fixtureName);
+
+            let duration;
+            if (test.startTime && test.endTime) {
+                duration = test.endTime - test.startTime;
+            }
+
+            const testStatus = (test.errs && test.errs.length) ? Status.Failed : Status.Passed;
+            suite.status = suite.status === Status.Passed ? testStatus : suite.status;
+            fixture.status = fixture.status === Status.Passed ? testStatus : fixture.status;
+
+            const addedTest = fixture.withTest(
+                test.name,
+                testStatus,
+                duration,
+                '',
+                test.startTime,
+                { browser: test.browser.name },
+            );
+
+            if (test.video) {
+                addedTest.attach({
+                    name: `${test.name}`,
+                    contentType: 'video/mp4',
+                    path: `${test.name}.mp4`,
+                });
+            }
+            for (const screenshot of test.screenshots) {
+                const suffix = path.basename(screenshot.screenshotPath);
+                const filename = `${test.name} - ${suffix}`;
```

Comment:
Its important that the filename here matches the name its given in the `getScreenshots` function. Might be worth extracting it to a function. And an integration test if you feel its both warranted and within the scope of this issue.